OrbitQt/servicedeploymanager.cpp

```diff
@@ -384,6 +421,12 @@ void ServiceDeployManager::StartWatchdog() {
 }
 
 outcome::result<ServiceDeployManager::GrpcPort> ServiceDeployManager::Exec() {
+  outcome::result<GrpcPort> result = outcome::success(GrpcPort{0});
+  DeferToBackgroundThreadAndWait(this, [&]() { result = ExecImpl(); });
+  return result;
+}
+
+outcome::result<ServiceDeployManager::GrpcPort> ServiceDeployManager::ExecImpl() {
```

Comment:
Do you want to add CHECKS here to make sure this is always called on background thread? just in case...

Reply:
Yes. That's already done but not pushed yet. There is also another problem on Linux which needs to be fixed before merging that! Thanks.

Reply:
Linux is now also working. QEventLoop didn't like to be created in a different thread. By deferring its initialization to the background thread the problem is gone. Might be a bug or an insufficiently communicated precondition of the class.